lib/spack/env/cc

```diff
@@ -342,7 +342,15 @@ case "$mode" in
         args=("${args[@]}" ${SPACK_LDLIBS[@]}) ;;
 esac
 
-full_command=("$command" "${args[@]}")
+#ccache only supports C languages, so filtering out Fortran
+if [[ ${lang_flags} != "F" && ${SPACK_CCACHE_BINARY} ]]; then
```

Comment:
Could we whitelist vs. blacklist here? e.g. `if ${lang_flags} = "C" || ${lang_flags} = "CXX"` (i probably didn't get that syntax right).

I'm aware that C/CXX/F are the only possible values currently but it's generally less fragile to specify the cases where it should enable ccache.